autoload/ale/highlight.vim

```diff
@@ -58,7 +58,7 @@ function! ale#highlight#RemoveHighlights() abort
 endfunction
 
 function! ale#highlight#UpdateHighlights() abort
-    let l:item_list = get(b:, 'ale_enabled', 1) && g:ale_enabled
+    let l:item_list = get(b:, 'ale_enabled', g:ale_enabled)
```

Comment:
Leave the `&& g:ale_enabled` check in here. ALE can't be enabled locally when it's disabled globally, so that we know we can remove the events needed for running locally when ALE is disabled globally.